src/stream/src/executor/source/executor_core.rs

```diff
@@ -35,7 +37,7 @@ pub struct StreamSourceCore<S: StateStore> {
 
     /// Split info for stream source. A source executor might read data from several splits of
     /// external connector.
-    pub(crate) latest_split_info: HashMap<SplitId, SplitImpl>,
+    pub(crate) latest_split_info: Arc<RwLock<HashMap<SplitId, SplitImpl>>>,
```

Comment:
We may try using the non-async `parking_lot::RwLock` to ensure that any stream or future will not capture the `RwLockGuard` (hold the guard across await point). If a stream or future capture the lock guard by accidence, the stream or future won't be `Send` and we will fail to pass the compiler check. Holding the lock guard across await point can easily cause deadlock, and from my understanding, in our current code, there is no code that we must hold the lock guard to call async method.

Reply:
> We may try using the non-async parking_lot::RwLock to ensure that any stream or future will not capture the RwLockGuard (hold the guard across await point).

Tried `parking_lot::RwLock` and it fails to compile and I switched to tokio locks. 

> in our current code, there is no code that we must hold the lock guard to call async method.

make sense, let me open another pr for it.